File: 백준/Silver/5635. 생일/생일.cc
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    int n, d, m, y;
    cin >> n;
    vector<vector<int>> date;
    vector<string> name(n);
    for (int i = 0; i < n; i++) {
        cin >> name[i] >> d >> m >> y;
        date.push_back({y, m, d, i});
    }
    sort(date.begin(), date.end());
    cout << name[date[date.size() - 1][3]] << "\n";
    cout << name[date[0][3]];
    return 0;
}

File: 백준/Silver/5635. 생일/생일.py
import sys
input=lambda:sys.stdin.readline().rstrip()
date=[]
for _ in range(int(input())):
    name,d,m,y=input().split()
    d,m,y=map(int,(d,m,y))
    date+=[(y,m,d,name)]
date.sort()
print(date[-1][3],date[0][3],sep="\n")